File: Alura/CursoAndroidTdd/app/src/main/java/br/com/alura/leilao/model/Lance.java
package br.com.alura.leilao.model;

import android.support.annotation.NonNull;

import java.io.Serializable;
import java.math.BigDecimal;

public class Lance implements Serializable, Comparable<Lance> {

    private final Usuario usuario;
    private final BigDecimal valor;

    public Lance(Usuario usuario, BigDecimal valor) {
        this.usuario = usuario;
        this.valor = valor;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public BigDecimal getValor() {
        return valor;
    }

    @Override
    public int compareTo(@NonNull Lance lance) {
        return lance.getValor().compareTo(valor);
    }
}

File: alura/curso_tdd_com_c_sharp/CursoTddComSharpAlura.Domain/Lance.cs
namespace CursoTddComCSharpAlura.Domain
{
    public class Lance
    {
        public Usuario Usuario { get; }

        public decimal Valor { get; }

        public Lance(Usuario usuario, decimal valor)
        {
            Usuario = usuario;
            Valor = valor;
        }
    }
}
